keel-bakery/src/main/kotlin/com/netflix/spinnaker/keel/bakery/resource/ImageHandler.kt

```diff
@@ -0,0 +1,135 @@
+package com.netflix.spinnaker.keel.bakery.resource
+
+import com.fasterxml.jackson.databind.ObjectMapper
+import com.netflix.spinnaker.igor.ArtifactService
+import com.netflix.spinnaker.keel.ami.BaseImageCache
+import com.netflix.spinnaker.keel.api.ArtifactType.DEB
+import com.netflix.spinnaker.keel.api.DeliveryArtifact
+import com.netflix.spinnaker.keel.api.Resource
+import com.netflix.spinnaker.keel.api.ResourceKind
+import com.netflix.spinnaker.keel.api.ResourceName
+import com.netflix.spinnaker.keel.api.SPINNAKER_API_V1
+import com.netflix.spinnaker.keel.bakery.api.ImageSpec
+import com.netflix.spinnaker.keel.clouddriver.CloudDriverService
+import com.netflix.spinnaker.keel.events.TaskRef
+import com.netflix.spinnaker.keel.model.Job
+import com.netflix.spinnaker.keel.model.OrchestrationRequest
+import com.netflix.spinnaker.keel.model.OrchestrationTrigger
+import com.netflix.spinnaker.keel.orca.OrcaService
+import com.netflix.spinnaker.keel.persistence.ArtifactRepository
+import com.netflix.spinnaker.keel.plugin.ResolvableResourceHandler
+import com.netflix.spinnaker.keel.plugin.ResourceDiff
+import com.netflix.spinnaker.keel.plugin.ResourceNormalizer
+import kotlinx.coroutines.runBlocking
+import org.slf4j.Logger
+import org.slf4j.LoggerFactory
+
+class ImageHandler(
+  override val objectMapper: ObjectMapper,
+  override val normalizers: List<ResourceNormalizer<*>>,
+  private val cloudDriver: CloudDriverService,
+  private val artifactRepository: ArtifactRepository,
+  private val baseImageCache: BaseImageCache,
+  private val orcaService: OrcaService,
+  private val igorService: ArtifactService
+) : ResolvableResourceHandler<ImageSpec, Image> {
+
+  override val apiVersion = SPINNAKER_API_V1.subApi("bakery")
+  override val supportedKind = ResourceKind(
+    apiVersion.group,
+    "image",
+    "images"
+  ) to ImageSpec::class.java
+
+  override fun generateName(spec: ImageSpec): ResourceName =
+    ResourceName("bakery:image:${spec.artifactName}")
+
+  override fun desired(resource: Resource<ImageSpec>): Image =
+    with(resource.spec) {
```

Comment:
In the future, do you image this could support a strategy? Like the spec might say "latest" or specify a version, and then the desired would resolve to the actual version?

Reply:
I _think_ so. I haven't given it a whole lot of thought.